index.bs

```diff
@@ -188,6 +188,10 @@ interface FileSystemHandle {
   readonly attribute FileSystemHandleKind kind;
   readonly attribute USVString name;
 
+  Promise<undefined> move(USVString new_entry_name);
+  Promise<undefined> move(FileSystemDirectoryHandle destination_directory);
+  Promise<undefined> move(FileSystemDirectoryHandle destination_directory, USVString new_entry_name);
```

Comment:
Hmm, this results in somewhat convoluted overloading I thought we were trying to move away from. @domenic what do you think? Should this use a dictionary?

Reply:
Also generally we don't follow Python and instead use camelCase for variable names.

Reply:
I think the overloading might be OK here, although you could reduce it by combining the latter two into `move(FileSystemDirectoryHandle destinationDirectory, optional USVString newEntryName)`.

In terms of API design, I think all of the following are worth considering:

- Positional parameters, varying order (this proposal):
  - `fileHandle.move("renamed.txt")`
  - `fileHandle.move(directoryHandle)`
  - `fileHandle.move(directoryHandle, "renamed.txt")`
- Positional parameters, consistent order (this will choke on https://github.com/plinss/widlparser/issues/79 but is still valid...)
  - `fileHandle.move("renamed.txt")`
  - `fileHandle.move(undefined, directoryHandle)`
  - `fileHandle.move("renamed.txt" directoryHandle)`
- Directory in an options bag (still requires overloading)
  - `fileHandle.move("renamed.txt")`
  - `fileHandle.move({ directory: directoryHandle })`
  - `fileHandle.move("renamed.txt", { directory: directoryHandle })`
- Options bag for everything
  - `fileHandle.move({ name: "renamed.txt"} )`
  - `fileHandle.move({ directory: directoryHandle })`
  - `fileHandle.move({ name: "renamed.txt", directory: directoryHandle })`
- Separate out move within directory vs. move to a new directory
  - `fileHandle.move("renamed.txt")` (or maybe `fileHandle.rename("renamed.txt")`)
  - `fileHandle.moveTo(directoryHandle)`
  - `fileHandle.moveTo(directoryHandle, "renamed.txt")`
- Path-based version?? Seems counter to how much of the rest of the API works, but is more what I would expect from other FS APIs.
  - `fileHandle.move("renamed.txt")`
  - `fileHandle.move(directoryHandle.resolve(fileHandle.name))` (not great...)
  - `fileHandle.move(directoryHandle.resolve("renamed.txt"))`

On balance I think all of these are pretty reasonable, including the current proposal.

Apologies if this has been discussed before; I didn't see it in a quick skim of the several issues and PRs that seem to be about this feature.

Reply:
On Chromium we already support `move()` for files within the OPFS, so if there's not a strong preference towards any one option I'd prefer to not make any breaking changes. Combining the latter two to make `newEntryName` optional seems reasonable, though, and wouldn't be a breaking change.

If we're convinced we need to change this, the options bag would be my preference since it's more extensible if we want to add more options in the future (e.g. "should this overwrite an existing file at this location?").

Regarding the `rename()` and `moveTo` proposals, I had previously proposed a separate `rename()` method for the first case, (i.e. `fileHandle.rename("renamed.txt")`) but received feedback from developers that a consolidated `move()` method was preferred.

I would especially recommend against a path-based version, since I think it could be confusing to only support relative paths and absolute paths don't make sense outside of the OPFS and won't make sense once sites have access to multiple OPFS instances when Storage Buckets is eventually supported. I'd much rather make it easier to get the destination directory through a convenience method like the one proposed here https://github.com/WICG/file-system-access/issues/378

Reply:
It's not the most pretty, but it seems reasonable enough to me